Adds bulk code generation to staff-dashboard

jkachel opened the pull request:
#### Pre-Flight checklist

- [X] Screenshots and design review for any changes that affect layout or styling
  - [X] Desktop screenshots
- [X] Migrations
  - [X] Migration is backwards-compatible with current production code
- [X] Testing
  - [X] Code is tested
  - [X] Changes have been manually tested

#### What are the relevant tickets?

Closes #1623

#### What's this PR do?

* Ups the maximum discount code length to 100 characters (since bulk codes use UUIDs, we were left with 13 characters for the bulk code prefix, which has been sort of a problem)
* Refactors bulk code generation into `api.py`
* Adds a Create Bulk Discount form to the staff dashboard to allow staff users to create codes on their own
   * The form works like the existing Create Discount form, with a couple of small modifications to support bulk generation
   * Generated codes are displayed after creation and the option is given to download the codes as a CSV file

#### How should this be manually tested?

Note that both your watch and refine containers should probably be restarted for testing this as there are new JavaScript dependencies. 

1. Load the staff dashboard and go to Discounts.
2. Click the Bulk Create button. 
3. Fill out the form appropriately. 
   4. Note that the prefix does not automatically append a separator between it and the code itself, so if you want one (our convention is to use a `-`) you need to specify it explicitly.
   5. If you fill out the date fields, make sure to also specify a time. Times are in UTC.
6. Click Save to create.
7. You should see the list of discount codes. 
8. Click Download CSV to download the generated codes in a CSV file.
9. The CSV file should be correct.

#### Screenshots (if appropriate)

![Screenshot 2023-05-24 at 12 36 38 PM](https://github.com/mitodl/mitxonline/assets/945611/ab1c8be0-2ee3-466f-b8e5-e2402d4751c1)
![Screenshot 2023-05-24 at 12 36 04 PM](https://github.com/mitodl/mitxonline/assets/945611/4ebca8d5-05be-4b9e-b2b9-d50967f85371)
![Screenshot 2023-05-24 at 12 35 27 PM](https://github.com/mitodl/mitxonline/assets/945611/d11049bc-4858-4239-8e50-e36b6928cd48)

(I changed the button title from Create Batch to Bulk Create for consistency but did not redo the screenshot.)

rachellougee reviewed (commented):
Can we add error handlings on the front end? I've received 400 error when missing a field, I can see `This field may not be blank` in the response if I inspect, it just doesn't show in the frontend. I've also got 500 error if I put 1 as Codes to Create. I guess whoever use it would not just create 1 code, but it would be great if we can handle 500 error and display missing field error messages.

Should we name "Codes to Create" to "Number of Codes to Create" to be clear?

It works as expected if everything fill up correctly.

jkachel commented:
Added some validation rules in https://github.com/mitodl/mitxonline/pull/1631/commits/6b48b1e037f30da518f2ce46ce85c96842e5ca10 . I updated the discount code count field to require a minimum of 2 - if you're only making 1 you should just be using the regular form, I think; it wasn't worth refactoring the api for this to accommodate it.

rachellougee reviewed (approved):
LGTM. Validation rules work as expected.
